bgpd/bgp_bmp.c

```diff
@@ -271,29 +312,43 @@ static void bmp_per_peer_hdr(struct stream *s, struct peer *peer,
 	stream_putc(s, flags);
 
 	/* Peer Distinguisher */
-	memset (&peer_distinguisher[0], 0, 8);
-	stream_put(s, &peer_distinguisher[0], 8);
+	stream_put(s, (uint8_t *)&peer_distinguisher, 8);
 
 	/* Peer Address */
-	if (peer->su.sa.sa_family == AF_INET6)
-		stream_put(s, &peer->su.sin6.sin6_addr, 16);
-	else if (peer->su.sa.sa_family == AF_INET) {
+	/* Set to 0 if it's a LOC-RIB INSTANCE (RFC 9069) or if it's not an
+	 * IPv4/6 address
+	 */
+	if (is_locrib || (peer->su.sa.sa_family != AF_INET6 &&
+			  peer->su.sa.sa_family != AF_INET)) {
 		stream_putl(s, 0);
 		stream_putl(s, 0);
 		stream_putl(s, 0);
-		stream_put_in_addr(s, &peer->su.sin.sin_addr);
```

Comment:
Shouldn't this be `stream_putl(s, 0);`? And V flag should be cleared too.

```
Zero-filled. The remote peer address is not applicable. The V flag is not applicable with the Loc-RIB Instance Peer Type considering addresses are zero-filled.
```

Reply:
> Shouldn't this be stream_putl(s, 0);?

it should and shouldn't :) there is [ongoing discussion at IETF](https://datatracker.ietf.org/doc/draft-francois-grow-bmp-loc-peer/) because not including this peer address causes correlation issues at the collectors.

i agree that since the RFC, for now, states that it is zero-filled, it should (only for loc-rib though).

> And V flag should be cleared too.

Do you mean for loc-rib ? If so then yes agreed. If not this is already done a few lines above:
https://github.com/FRRouting/frr/blob/c4e030ac87687b15418df652f69e3959c58de0e2/bgpd/bgp_bmp.c#L266-L271